RIOT-OS/RIOT
# doc: fix references and inches unit

### Contribution description

This PR fixes the references and inches unit symbols in documentation of `{boards,cpu,drivers}.

Doxygen version 1.9.4 that is used on https://doc.riot-os.org doesn't support the following any longer:
- References with line breaks between `[...]` and `(...)` which lead either to
  - missing images (e.g. in https://doc.riot-os.org/group__boards__nucleo-f030r8.html) or to
  - source code instead of references (e.g. in https://doc.riot-os.org/group__boards__deRFmega128.html)
- Using `"` as inch unit wihtout `\` which lead to complete messed up documentatoin
  (e.g. in https://doc.riot-os.org/group__boards__stm32f746g-disco.html)

### Testing procedure

Use doxygen 1.9.4 and `make doc` to generate the documentation and check the examples for correctness.

### Issues/PRs references
